### ISSUE TITLE
`aria-disabled` buttons should have same CSS as `disabled`

### ISSUE THREAD

**Issue by coliff** [twbs/bootstrap#39727]
If you're building a form then you may want the submit button to be `disabled` until the form has been completed correctly. Using the `disabled` attribute on the submit button will however not allow the user to tab to it.

A better solution could to be disable click events on the submit button and make the button be announced as disabled with the `aria-disabled="true"` attribute. If you do that with Bootstrap though it doesn't use the disabled state button CSS. (Quick demo: https://stackblitz.com/edit/te5a7k?file=index.html )

I read about this on: https://a11y-101.com/development/aria-disabled

TLDR: A disabled button currently has CSS something like this:

```css
.btn.disabled, .btn:disabled, fieldset:disabled .btn {
   ....
}
```

I suggest it should include `aria-disabled="true"` buttons so compiled CSS would be:

```css
.btn.disabled, .btn:disabled, fieldset:disabled .btn, .btn.aria-disabled="true" {
    ....
}
```

I know there are more accessibility considerations regarding my example above and it may not be the optimal way, but I still think it probably makes sense for `aria-disabled="true"` buttons to inherit styles from `disabled` buttons.

**Comment by julien-deramond** [comment]
Thanks for this suggestion @coliff. Your StackBlitz link seems to be broken, could you please update it if possible?

If I understand the content correctly, we're talking here simply to ensure some visual consistency so that `disabled` or `aria-disabled="true"` applied on a button would have the same rendering. My first feeling would be that it could work even if we would need to double-check some edge cases.

@patrickhlauke would it make sense to you to have such a defensive CSS approach or would it be necessary on the contrary to have a different rendering?

**Comment by AgentKaash** [comment]
Could you @coliff  or @julien-deramond tell me, in which section of Bootstrap should i find the actual issue in real time, I opend Bootstrap many times but I couldn't find it.
 I want to fix it. My humble request please give me some  hints....
 Though I am newbie in contribution I am going anywhere whithout solve it.

**Comment by julien-deramond** [comment]
Thanks @AgentKaash but this is not yet decided whether we're going to make this modification or not. So there's no PR to provide at the moment.

> in which section of Bootstrap should i find the actual issue in real time, I opend Bootstrap many times but I couldn't find it

It's not in the documentation. This would be a defensive CSS approach for this use case.

**Comment by AgentKaash** [comment]
for me , @julien-deramond , i really want to. anyways thanks for helping me out . But i figure out many things .Though this is not for me but i learnt something new today as a newbie in contribution. 
Sir would you suggest me some of projects? i know you all are busy somewhere for something new, if you help me i will very greatful.

and thanks a lot sir. wish you have a wonderful day.

**Comment by patrickhlauke** [comment]
In principle, I don't have anything against also applying the regular `.disabled` styling to cases where `aria-disabled="true"` is set. 
Of course, you can achieve this currently by manually adding the `.disabled` class (like you would with "faked" buttons that use `<a>`, for instance)

```html
<button type="button" class="btn btn-primary disabled" aria-disabled="true"> … </button>
```

*If* we go with these styles being applied out-of-the-box directly based on the `aria-disabled` attribute, we can then tweak the explanation in https://getbootstrap.com/docs/5.3/components/buttons/#disabled-state about authors needing to add `.disabled` themselves.

If we *don't* want to go with this, we can at least *expand* the explanation in https://getbootstrap.com/docs/5.3/components/buttons/#disabled-state to mention that the same caveat about adding `.disabled` to links also needs to be done for buttons that aren't *really* `disabled`

**Comment by developernarendra** [comment]
is it something which i can solve?

**Comment by patrickhlauke** [comment]
We'd probably need to make a decision on which direction to go (see my two suggestions above). @julien-deramond et al?

**Comment by mdo** [comment]
@coliff if this is still of interest to you, feel free to whip up a PR for us to review. For now, I'm going to close as a won't fix (not a hard no, just not something I care enough to do but wouldn't say no to someone else doing).
